ir_actions_report_xml_multi_company/models/ir_actions_report_xml.py

```diff
@@ -0,0 +1,18 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class IrActionsReportXml(models.Model):
+
+    _inherit = 'ir.actions.report.xml'
+
+    @api.model
+    def _default_company(self):
+        return self.env['res.company']._company_default_get('res.partner')
+
+    company_id = fields.Many2one(
+        comodel_name='res.company',
+        string='Company', default=_default_company)
```

Comment:
By installing this module on an already existing multicompany database, all the existing reports are automatically set with the default Company. This way all the existing reports become hidden for all the other companies. The point is that the most of the reports already present in the database are supposed to be used by all the companies; only few reports (eg. the invoices) are usually customized per company.

I would personally prefer to NOT define the default company at all.
When the module is installed, the administrator should update the per-company reports with the proper Company.